Sources/SwifterSwift/SwiftStdlib/CollectionExtensions.swift

```diff
@@ -89,26 +81,47 @@ public extension Collection where Index == Int {
     ///   - slice: size of array in each interation.
     ///   - body: a closure that takes an array of slice size as a parameter.
     func forEach(slice: Int, body: ([Element]) throws -> Void) rethrows {
-        guard slice > 0, !isEmpty else { return }
+        guard slice > 0 else { return }
 
-        var value: Int = 0
-        while value < count {
-            try body(Array(self[Swift.max(value, startIndex)..<Swift.min(value + slice, endIndex)]))
-            value += slice
```

Comment:
What about 
```swift
var start = startIndex
repeat {
     let end = index(start, offsetBy: slice, limitedBy: endIndex) ?? endIndex
     try body(Array(self[start..<end]))
     start = end
} while (end != endIndex)
```
Suggestion, not sure this compiles(just sketch here)/ or handles all the cases, but I just searching for a way to avoid/simplify the last check and the duplicated index :)

Reply:
Maybe that would break on empty collection ...

Reply:
It doesn't compile because `end` is declared inside the loop so it isn't recognised outside the scope. I found a solution though :)

Reply:
Nice :)